libsubmit/providers/local/local.py

```diff
@@ -92,7 +92,13 @@ def __init__ (self, config, channel_script_dir=None, channel=None):
              - Config (dict): Dictionary with all the config options.
         '''
 
-        self.channel = channel
+        if channel is None:
+            if channel_script_dir is None:
+                self.channel = LocalChannel(scriptDir=channel_script_dir)
+            else:
+                self.channel = LocalChannel()
+        else:
+            self.channel = channel
```

Comment:
Maybe put a note about why we create the channel in the provider where the general pattern is to have the 
provider be passed a channel from the execution provider factory.